src/providers/aws.rs

```diff
@@ -1369,6 +1278,112 @@ impl RegionLauncher {
 
         Ok(())
     }
+
+    #[instrument(level = "debug")]
+    async fn describe_spot_instance_requests(
+        &self,
+    ) -> Result<Vec<(String, String, Option<String>)>, Report> {
+        let client = self.client.as_ref().unwrap();
+        let request_ids = self.spot_requests.keys().cloned().collect();
+        let mut req = rusoto_ec2::DescribeSpotInstanceRequestsRequest::default();
+        req.spot_instance_request_ids = Some(request_ids);
+        loop {
+            let res = client.describe_spot_instance_requests(req.clone()).await;
+            if let Err(ref e) = res {
+                let msg = e.to_string();
+                if msg.contains("The spot instance request ID") && msg.contains("does not exist") {
+                    tracing::trace!("spot instance requests not yet ready");
+                    continue;
```

Comment:
Will this hammer the api? Should we add a delay?

Reply:
> One question: there is now a nested-`loop{}` for `wait_for_spot_instance_requests -> describe_spot_instance_requests`. Is this intentional? The flow changes from checking all the spot requests at once, to checking one at a time?

I believe the flow didn't change. However, this loop doesn't have a timeout. From my limited experience it's not a problem in practice, but maybe this loop should timeout as well?

Reply:
Ah, I see now :)
I think the only case where the inner loop would loop forever is if the spot instance request ID never showed up? I agree it's probably safe to assume the aws api will report its existence eventually, since `make_spot_instance_requests` succeeded at this point.